algorithms/src/traits/signature.rs

```diff
@@ -28,17 +24,7 @@ pub trait SignatureScheme:
     Sized + ToBytes + FromBytes + Debug + Clone + Eq + Send + Sync + From<<Self as SignatureScheme>::Parameters>
 {
     type Parameters: Clone + Debug + Eq;
-    type PublicKey: Clone
-        + Debug
-        + Default
-        + ToBytes
-        + FromBytes
-        + Hash
-        + Eq
-        + Send
-        + Sync
-        + CanonicalSerialize
-        + CanonicalDeserialize;
```

Comment:
Note: Since DPC no longer uses `CanonicalSerialize` and `CanonicalDeserialize`, they are removed from the trait.